src/install/deploySteps.js

```diff
@@ -156,42 +101,40 @@ module.exports = (apps, mode, deployDoc) => {
       });
   };
 
-  const updateSymlinkAndRemoveOldVersion = changedApps => {
+  const updateSymlink = changedApps => {
     return Promise.all(changedApps.map(app => {
-      const livePath = deployPath(app, 'current');
+      const livePath = app.deployPath('current');
 
       if(fs.existsSync(livePath)) {
         const linkString = fs.readlinkSync(livePath);
 
         if(fs.existsSync(linkString)) {
-          debug(`Deleting old ${app.name} from ${linkString}…`);
-          fs.removeSync(linkString);
+          fs.symlinkSync(linkString, app.deployPath('old'));
         } else debug(`Old app not found at ${linkString}.`);
 
         fs.unlinkSync(livePath);
       }
 
-      fs.symlinkSync(deployPath(app), livePath);
+      fs.symlinkSync(app.deployPath(), livePath);
     }));
   };
 
   const processDdoc = (ddoc, firstRun) => {
```

Comment:
can we rename the `ddoc` var to `wrappedDdoc` or something? Since we are without types it's confusing to call it a ddoc when it's not like the rest of the ddoc variables we are pushing around.